### ISSUE TITLE
Opening a MongoDB collection per request is NOT recommeded

### ISSUE THREAD

**Issue by nick13jaremek** [angelsanchez/nodejs-atdd-example#5]
Hi there!

The application on this repository calls the _MongoDB_ `connect` method on each find, update, delete... action. Such a practice is not correct due to two main reasons:
1. The _MongoDB_ authors **recommend calling MongoClient.connect once and reusing the database variable returned by the callback**, [as stated in their docs](https://mongodb.github.io/node-mongodb-native/driver-articles/mongoclient.html)
2. Each _MongoDB_ `connect` call requires a new system's file descriptor. This means that each request to the application (such as the one in this repository) will increase the system's open file descriptors. The Ubuntu distribution has a limit of open file descriptors, which defaults to 1024 or similar (this number can be checked by running the `ulimit -n` command on a Linux distribution). 

Although the most simple solution would be increasing the limit of open file descriptors, it is not a definitive solution. When reaching the limit of open file descriptors, the application will fail for new incoming requests due to an _EMFILE: Too many files open" *MongoDB_ error.

Therefore:
_The best solution would be following the guidelines of the MongoDB authors, that is, connect once and reuse the Mongo connection._

**TL;DR**

**Do not connect to the MongoDB database on each incoming request, but connect once and reuse this connection for subsequent requests to MongoDB**

**Comment by guumaster** [comment]
We though that we may solve it with a Singleton DB object and a `on-finished` middleware to close it. So you only have the amount of opened connections as concurrent requests.

Or, as @nick13jaremek mentions, we could go back to one connection per service and handle disconnection events. I like best the singleton approach.
